std/random.d

```diff
@@ -1336,6 +1330,297 @@ version (AnyARC4Random)
+        // in all situations where you do not require a persisted key.
+        // CRYPT_SILENT is intended for use with applications for which the UI
+        // cannot be displayed by the CSP.
+        if (!(fnCryptAcquireContextW && fnCryptAcquireContextW(
+                &hProvider, null, null,
+                PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT)))
+        {
+            if (GetLastError() == NTE_BAD_KEYSET)
+            {
+                // Attempt to create default container
+                auto fnCryptAcquireContextA = cast(FnCryptAcquireContextW)
+                    GetProcAddress(hAdvapi32, "CryptAcquireContextA");
+                if (!(fnCryptAcquireContextA && fnCryptAcquireContextA(
+                    &hProvider, null, null,
+                    PROV_RSA_FULL, CRYPT_NEWKEYSET | CRYPT_SILENT)))
+                {
+                    return -1;
+                }
+            }
+        }
+        assert(hProvider);
+        scope (exit) fnCryptReleaseContext(hProvider, 0);
+        auto fnCryptGenRandom = cast(FnCryptGenRandom)
+            GetProcAddress(hAdvapi32, "CryptGenRandom");
+        if (!fnCryptGenRandom) return -1;
+        if (fnCryptGenRandom(hProvider, len, cast(PBYTE) ptr))
+            return len;
+        else
+            return -1;
+    }
+}
+else version (Posix)
+{
+    version (linux)
+    {
+        version (X86) private enum NR_getrandom = 355;
+        else version (X86_64) private enum NR_getrandom = 318;
+        else version (ARM) private enum NR_getrandom = 384;
+        else version (AArch64) private enum NR_getrandom = 278;
+        else version (SPARC) private enum NR_getrandom = 347;
+        else version (SPARC64) private enum NR_getrandom = 347;
+        else version (Alpha) private enum NR_getrandom = 511;
+        else version (IA64) private enum NR_getrandom = 1339;
+        else version (PPC) private enum NR_getrandom = 359;
+        else version (PPC64) private enum NR_getrandom = 359;
+        else version (S390) private enum NR_getrandom = 349;
+        else version (SystemZ) private enum NR_getrandom = 349;
+        else version (HPPA) private enum NR_getrandom = 339;
+        else version (HPPA64) private enum NR_getrandom = 339;
+        else version (MIPS_O32) private enum NR_getrandom = 4353;
+        else version (MIPS_N32) private enum NR_getrandom = 6317;
+        else version (MIPS64) private enum NR_getrandom = 5313;
+        else version (SH)
+        {
+            static if (size_t.sizeof == 4)
+                private enum NR_getrandom = 373;
+            else static if (size_t.sizeof == 8)
+                private enum NR_getrandom = 384;
+        }
+
+        static if (is(typeof(NR_getrandom)))
+        {
+            // getrandom was introduced in Linux 3.17.
+            // checks whether the Linux kernel supports getRandom by looking at the
+            // reported version.
+            private auto checkHasGetRandom()() @nogc nothrow @trusted
```

Comment:
Rather than an explicit check it might make more sense to just try the syscall and check for an error code.

Reply:
When I implemented this, I went with this because it's more "bullet-proof" than hoping that the kernel provides a proper error code, but yeah it should work fine too.